src/engine.rs

```diff
@@ -904,20 +943,19 @@ impl Reedline {
                                         prompt_offset.1 -= 1;
                                     }
                                 } else {
-                                    self.run_edit_commands(&[EditCommand::InsertChar(c)]);
+                                    self.run_edit_commands(&[InsertChar(c)]);
                                 }
+                                self.line_buffer.set_previous_lines();
```

Comment:
Have you attempted this with history search? I assume there is a potential unwanted interaction, but the traditional reverse search is broken enough that debugging the interaction is lower priority

Reply:
> test broken |

@sherubthakur  Thanks for the great test, in fact I have not considered this case where the number of words is the same but the content for some reason has changed

Reply:
No I didn't test with history sorry

Reply:
From my understanding of the history logic now, this line should not cause an issue. But you probably want to think about how you incorporate the change after recalling something from history.